core/templates/domain/recommendations/exploration-recommendations-backend-api.service.ts

```diff
@@ -26,12 +26,15 @@ import {
   LearnerExplorationSummaryObjectFactory
 } from 'domain/summary/learner-exploration-summary-object.factory';
 
+// This is the type used for paramas that are sent to the backend.
```

Comment:
typo: params (ditto two lines below, too).

I don't understand the justification for optional properties. Can't we fully fix the list of properties, using techniques similar to those mentioned in my previous comments above?

Reply:
I don't think so. Because for example when I tried to do that the URL is something like `?collection_id=null` and the backend receives the collection_id as "null" and starts to look for a collection with id `"null"` and which in return gives an error.

Reply:
Ah, I see what you mean, thanks for clarifying. Could you clarify the comment please? Currently it's very vague "may not be applicable" and I think you want to replace that with something more concrete like "may not be present in the URL". You can also add a note that, otherwise, we will end up sending a url like `?collection_id=null` which will not be recognized by the backend.

Reply:
done